baybe/surrogates/base.py

```diff
@@ -56,112 +100,261 @@ class Surrogate(ABC, SerialMixin):
     """Class variable encoding whether or not the surrogate supports transfer
     learning."""
 
-    def to_botorch(self) -> Model:
-        """Create the botorch-ready representation of the model."""
+    _searchspace: SearchSpace | None = field(init=False, default=None, eq=False)
```

Comment:
One thing that I just noticed: Now that we also have to keep a reference to the searchspace on which the surrogates was trained, having the `ModelContext` explicitly part of the signatures is actually no longer required, since the surrogates can still extract the necessary information directly from the object attributes. Not saying that we need to remove the concept per se, i.e. I think it's nice that the GP still cleanly encapsulates all relevant context information in one object that properly maintains and structures the information. But we *could* in fact remove the `ModelContext` again from the `_posterior` signature. What do you think? Perhaps it makes the class just leaner if we omit it?

Reply:
Do you mean `_fit` instead of `_posterior`? Because `_posterior` does not have that function as part of its signature.

Reply:
doesnt seem neessary but encapsualtes away the metainfo so im fine wiht keeping it

Reply:
Made a compromise: removed it from the signature (imposing less requirements is better) but kept it internally in the GP model, because the encapsulation is still nice 👍🏼